app/_hub/kong-inc/opa/index.md

```diff
@@ -0,0 +1,307 @@
+    - name: include_service_in_opa_input
+      required: false
+      datatype: boolean
+      default: false
+      description: |
+        If set to true, the Kong Service object in use for the current request is included as input to OPA.
+    - name: include_route_in_opa_input
+      required: false
+      datatype: boolean
+      default: false
+      description: |
+        If set to true, the Kong Route object in use for the current request is included as input to OPA.
+    - name: include_consumer_in_opa_input
+      required: false
+      datatype: boolean
+      default: false
+      description: |
+        If set to true, the Kong Consumer object in use for the current request (if any) is included as input to OPA.
+---
+
+## Usage
+
+### Set up an OPA server
+
+If you are not already running an OPA server, start one by following instructions
+on [openpolicyagent.org](https://openpolicyagent.org).
+
+### Set up a dummy policy
+
+Create an `example.rego` file with the following content:
+
+```rego
+package example
+
+default allow = false
+
+allowBoolean {
+  header_present
+}
+
+header_present {
+  input.request.http.headers["my-secret-header"] == "open-sesame"
+}
+
+allowDetailed = response {
+  header_present
+  response := {
+    "allow": true,
+    "headers": {
+      "header-from-opa": "accepted",
+    },
+  }
+}
+
+allowDetailed = response {
+  not header_present
+  response := {
+    "allow": false,
+    "status": 418,
+    "headers": {
+      "header-from-opa": "rejected",
+    },
+  }
+}
+```
+
+This policy allows the request to pass through if it contains the HTTP header
+`my-secret-header` with value `open-sesame`.
+
+
+Next, configure the policy with OPA:
+
+```bash
+curl -XPUT localhost:8181/v1/policies/example --data-binary @example.rego
+```
+
+The above command uses OPA's default port 8181. It could be different for your
+setup.
+
+### Set up Kong
+
+Set up a Route and Service in Kong and then enable the plugin:
+
+```bash
+curl -X POST http://<admin-hostname>:8001/routes/<route>/plugins \
+    --data "name=opa"  \
+    --data "config.opa_path=/v1/data/example/allow" \
+    --data "config.opa_host=<host-where-opa is running>"
+```
+
+### Make a request
+
+Now, make a request to Kong:
+
+```bash
+curl http://kong:8000/{proxy_path}
+```
+
+You will get a 403 response from Kong because OPA has rejected the request.
+
+Next, make the same request providing the necessary header:
+
+```bash
+curl http://kong:8000/{proxy_path} \
+  -H 'my-secret-header: open-sesame'
+```
+
+This time you will get the response from your upstream service, as this request
+is allowed by OPA.
+
+Next, update the plugin configuration to use `/v1/data/example/allowDetailed` as
+`opa_path` and observe how the policy affects the response and proxied request.
+
+## OPA input structure
+
+The input to OPA has the following JSON structure:
+
+```json:
```

Comment:
This codeblock is extremely difficult to read in the output. I'd take the comments out and put them in a list after the codeblock.

Reply:
> This codeblock is extremely difficult to read in the output. I'd take the comments out and put them in a list after the codeblock.

@lena-larionova I don't think they are difficult to read in the output, but for GA, this would be a great opportunity to use the numbered code blocks and move the comments to a table underneath.  cc: @hbagdi

Reply:
Removing the `json` from the codeblock markdown definition might make the rendering better. The comments are contextual and make it easier for the reader to grasp the entire JSON as a whole.

Doing as a list below a codeblock is fine too but my personal preference is what we have.

Reply:
Bleh. Harry's right - removing the `json` class from the codeblock will be better; right now, the output looks horrible:
![Screen Shot 2021-04-13 at 10 21 46 AM](https://user-images.githubusercontent.com/54370747/114594409-13451080-9c42-11eb-9697-0b5e30e48551.png)